array.go

```diff
@@ -913,3 +913,58 @@ func countElement(array []interface{}, element interface{}) int {
 	}
 	return count
 }
+
+// Filter accepts a function that returns a boolean. The function is ran
+// over the array items. If the function returns true, the item passes
+// the filter and is added to the new array of filtered items. If false,
+// the item is skipped (or in other words filtered out). After iterating
+// through all the items of the original array, the new filtered array
+// is returned.
+//
+// If there are any failed assertions in the filtering function, the
+// item is omitted without causing chain failure.
+//
+// Example:
+//
+// array := newArray(reporter, []interface{}{1, 2, "foo", "bar"}
+// filteredArray := array.Filter(func(_ int, value *Value) bool {
```

Comment:
Let's replace `_ int` with `index int` to give a hint for the meaning of that argument.